File: 1557.cpp
struct Solution {
    vector<int> findSmallestSetOfVertices(int n, const vector<vector<int>>& edges) {
        vector<bool> arr(n, true);
        for (const vector<int> &e : edges){
            arr[e[1]] = false;
        }
        vector<int> ans;
        for (int i = 0; i < n; ++i){
            if (arr[i]){
                ans.push_back(i);
            }
        }
        return ans;
    }
};

File: sol/1557/1557.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        arr = [True] * n
        for e in edges:
            arr[e[1]] = False
        for i in range(n):
            if arr[i]:
                yield i
